node/tessel-export.js

```diff
@@ -1449,6 +1417,41 @@ function getWifiInfo() {
       });
     }
 
+    // when immediately connecting and restarting the wifi chip, it takes a few moments before an IP address is broadcast to Tessel.
+    // This function keeps checking for that IP until it's available.
+    function recursiveIP(network) {
+      setImmediate(() => {
+        childProcess.exec('ifconfig wlan0', (error, ipResults) => {
+          if (error) {
+            reject(error);
+          } else if (ipResults.includes('Bcast') === false) {
+            recursiveWifi(network);
+          } else {
+            network.ip = ipResults
+              .split('\n') // split multiple line string into array of strings
+              .filter((string) => string.includes('Bcast'))[0] // find the string containing a reference to "Bcast" (Broadcast IP)
+              .trim() // remove extra whitespace from either side of the string
+              .split(' ') // split the string by whitespace between words
+              .filter((string) => string.includes('Bcast'))[0] // find the string containing a reference to "Bcast"
+              .split(':')[1]; // split the string by the ":", i.e. "Bcast:10.0.0.11" and grab the number as the 2nd item in that array
```

Comment:
I think everything in lines 1425-1436 can be simplified...

```js
// This actually goes at the beginning of getWifiInfo()
var rbcast  = /(Bcast):([\w\.]+)/;
// ...

if (error) {
  reject(error);
} else {
  var bcastMatches = ipResults.match(rbcast);

  if (bcastMatches === null) {
    recursiveWifi(network);
  } else {
    // Successful matches will have a result that looks like: 
    // ["Bcast:0.0.0.0", "Bcast", "0.0.0.0"]
    if (bcastMatches.length === 3) {
      network.ip = bcastMatches[2];
    } else {
      recursiveWifi(network);
    }
  }
}
```




Here's the `ipResults` for my T2 that I tested the above on: 

```
wlan0     Link encap:Ethernet  HWaddr 02:A3:CE:A7:F4:68
          inet addr:192.168.1.3  Bcast:192.168.1.255  Mask:255.255.255.0
          inet6 addr: fe80::a3:ceff:fea7:f468/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:441 errors:0 dropped:0 overruns:0 frame:0
          TX packets:186 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:92366 (90.2 KiB)  TX bytes:29894 (29.1 KiB)
```